apps/cowswap-frontend/src/modules/permit/utils/handlePermit.ts

```diff
@@ -15,19 +13,24 @@ import { HandlePermitParams } from '../types'
  * Returns the updated appData
  */
 export async function handlePermit(params: HandlePermitParams): Promise<AppDataInfo> {
-  const { permitInfo, inputToken, provider, account, chainId, appData } = params
+  const { permitInfo, inputToken, account, appData, generatePermitHook } = params
 
   if (permitInfo) {
     // permitInfo will only be set if there's enough allowance
 
     const permitData = await generatePermitHook({
       inputToken,
-      provider,
       account,
-      chainId,
       permitInfo,
     })
 
+    if (!permitData) {
+      // Not able to generate permit data, remove it from the order
+      // TODO: maybe should throw instead?
+      console.warn(`[handlePermit] Unable to generate permit data`)
+      return updateHooksOnAppData(appData, undefined)
```

Comment:
I think so, if we let the user continue, the order will be created without the hook, so it will never execute